tripal/src/Hook/TripalHooks.php

```diff
@@ -0,0 +1,392 @@
+   * Tripal fields but which are attached to a tripal content type.
+   */
+  #[Hook('form_alter')]
+  public function formFieldConfigEditFormAlter(&$form, FormStateInterface $form_state, $form_id) {
+    if ($form_id === 'field_config_edit_form') {
+      $form_object = $form_state->getFormObject();
+      if (method_exists($form_object, 'getEntity')) {
+        /** @var \Drupal\field\Entity\FieldConfig $field **/
+        $field = $form_object->getEntity();
+        if (method_exists($field, 'getTargetEntityTypeId')) {
+          if ($field->getTargetEntityTypeId() == 'tripal_entity') {
+            $elements = TripalFieldItemBase::buildFieldTermForm($field, $form, $form_state);
+            $form['settings'] = $form['settings'] + $elements;
+          }
+        }
+      }
+    }
+  }
+
+  /**
+   * Implements hook_preprocess_field().
+   *
+   * This is to permit HTML markup in the page title field, for
+   * example on an organism page, to display genus and species
+   * in italics.
+   */
+  #[Hook('preprocess_field')]
+  public function preprocessField(&$variables) {
+    if ($variables['element']['#field_name'] == 'title') {
+      // We can configure which tags are allowed at /admin/tripal/config.
+      $tag_string = \Drupal::config('tripal.settings')->get('tripal_entity_type.allowed_title_tags');
+      $tripal_allowed_tags = explode(' ', $tag_string ?? '');
+
+      // Process each item (usually only a single one).
+      foreach ($variables['items'] as $delta => $item) {
+        // The title can be either a simple inline template or a link.
+        if ($item['content']['#type'] == 'inline_template') {
+          $value = $item['content']['#context']['value'];
+          // Convert strings into markup, this will cause HTML tags to
+          // be rendered.
+          if (is_string($value)) {
+            $sanitized_value = Xss::filter($value, $tripal_allowed_tags);
+            $variables['items'][$delta]['content']['#context']['value'] = Markup::create($sanitized_value);
+          }
+        }
+        elseif ($item['content']['#type'] == 'link') {
+          $value = $item['content']['#title']['#context']['value'];
+          // Convert strings into markup, this will cause HTML tags to
+          // be rendered.
+          if (is_string($value)) {
+            $sanitized_value = Xss::filter($value, $tripal_allowed_tags);
+            $variables['items'][$delta]['content']['#title']['#context']['value'] = Markup::create($sanitized_value);
+          }
+        }
+      }
+    }
+  }
+
+  /**
+   * Implements hook gin_content_form_routes provided by the GIN admin theme.
+   */
+  #[Hook('gin_content_form_routes')]
+  public function ginContentFormRoutes() {
+    return [
+      'entity.tripal_entity.add_form',
+      'entity.tripal_entity.edit_form',
+    ];
+  }
+
+  /**
+   * Implements hook_views_pre_view().
+   */
+  #[Hook('views_pre_view')]
+  public function viewsPreView(ViewExecutable $view, $display_id, array &$args) {
```

Comment:
I think these files could get quite long... maybe we should move this and viewPresave into a `TripalViewsHooks` class? Views are one of the modules that have the most hooks and I suspect we may be implementing more of them as we continue to work on the views performance.

Reply:
Commit https://github.com/tripal/tripal/pull/2356/commits/161f2d4e0d701694d7507a53bacbf5b9e3a78c64